app/models/task.rb

```diff
@@ -1362,6 +1363,20 @@ def read_file_from_done(idx)
     nil
   end
 
+  def send_feedback_notification
+    return unless task_status.in?([TaskStatus.redo, TaskStatus.fail, TaskStatus.fix_and_resubmit, TaskStatus.feedback_exceeded, TaskStatus.discuss, TaskStatus.demonstrate, TaskStatus.complete])
+    return unless project.student.receive_feedback_notifications
+    return unless unit.send_notifications
```

Comment:
`unit` could be null here if `project` does not have a `unit` as there is no constraint that `unit` cannot be null in the schema. Use `&` to safely output null without throwing null exception

```suggestion
    return unless unit&.send_notifications
```